tools/pubsys/src/aws/publish_ami/mod.rs

```diff
@@ -0,0 +1,459 @@
+//! The publish_ami module owns the 'publish-ami' subcommand and controls the process of granting
+//! and revoking public access to EC2 AMIs.
+
+use crate::aws::{client::build_client, region_from_string};
+use crate::config::InfraConfig;
+use crate::Args;
+use futures::future::{join, ready};
+use futures::stream::{self, StreamExt};
+use log::{debug, error, info, trace};
+use rusoto_core::{Region, RusotoError};
+use rusoto_ec2::{
+    DescribeImagesError, DescribeImagesRequest, DescribeImagesResult, Ec2, Ec2Client,
+    ModifyImageAttributeError, ModifyImageAttributeRequest, ModifySnapshotAttributeError,
+    ModifySnapshotAttributeRequest,
+};
+use snafu::{ensure, OptionExt, ResultExt};
+use std::collections::{HashMap, HashSet};
+use std::fs::File;
+use std::iter::FromIterator;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+/// Grants or revokes permissions to Bottlerocket AMIs
+#[derive(Debug, StructOpt)]
+#[structopt(setting = clap::AppSettings::DeriveDisplayOrder)]
+#[structopt(group = clap::ArgGroup::with_name("mode").required(true).multiple(false))]
+pub(crate) struct PublishArgs {
+    /// Path to the JSON file containing regional AMI IDs to modify
+    #[structopt(long)]
+    ami_input: PathBuf,
+
+    /// List of regions to publish in, overriding Infra.toml; must match regions in --ami-input
+    /// file.
+    #[structopt(long, use_delimiter = true)]
+    regions: Vec<String>,
+
+    /// Make the AMIs public
+    #[structopt(long, group = "mode")]
+    make_public: bool,
+    /// Make the AMIs private
+    #[structopt(long, group = "mode")]
```

Comment:
Nit: matches the spacing of the other arguments.
```suggestion
    make_public: bool,
    
    /// Make the AMIs private
    #[structopt(long, group = "mode")]
```

Reply:
public/private are tied and should be read together, that's why we kept them together without a blank line.